v2/tools/generator/internal/codegen/pipeline/prune_resources_with_lifecycle_owned_by_parent.go

```diff
@@ -69,19 +75,66 @@ func PruneResourcesWithLifecycleOwnedByParent(configuration *config.Configuratio
 	return stage
 }
 
+func flagPrunedEmptyProperties(defs astmodel.TypeDefinitionSet, emptyPrunedProps astmodel.TypeNameSet) (astmodel.TypeDefinitionSet, error) {
+	emptyObjectVisitor := astmodel.TypeVisitorBuilder{
+		VisitObjectType: tagEmptyObjectARMProperty,
+	}.Build()
+
+	emptyPrunedPropertiesArm := astmodel.NewTypeNameSet()
+	for emptyPrunedProp := range emptyPrunedProps {
+		// we need to add the noConversion tag on ARM type for the empty pruned property to relax the validation for convertToARM function.
+		armDef, err := GetARMTypeDefinition(defs, emptyPrunedProp)
+		if err != nil {
+			return nil, err
+		}
+		emptyPrunedPropertiesArm.Add(armDef.Name())
+	}
+
+	result, err := emptyObjectVisitor.VisitDefinitions(defs, emptyPrunedPropertiesArm)
+	if err != nil {
+		return nil, err
+
+	}
+
+	return result, nil
+}
+
 type misbehavingEmbeddedTypePruner struct {
-	configuration *config.Configuration
+	configuration         *config.Configuration
+	emptyPrunedProperties astmodel.TypeNameSet
 }
 
-func newMisbehavingEmbeddedTypeVisitor(configuration *config.Configuration) astmodel.TypeVisitor {
+func newMisbehavingEmbeddedTypeVisitor(configuration *config.Configuration) (astmodel.TypeVisitor, *misbehavingEmbeddedTypePruner) {
```

Comment:
hmm, maybe refactor this to just return `pruner` and have `pruner` have a property `visitor`?

It's a bit cyclic but would result in a single return type rather than two, which feels cleaner?

Reply:
Done!